### PR TITLE
Update readelf information with onnx-mlir commit

### PR DESCRIPTION
**Pull request by hamptonm1**
I updated the following based on the issue @cjvolzka created. 

If no vendor / product info is specified show the `onnx-mlir` information from when the compiler itself was built. For example:

```
String dump of section '.comment':
  [     0]  GCC: (GNU) 8.5.0 20210514 (Red Hat 8.5.0-15)
  [    2d]  onnx-mlir version 0.3.1 <onnx-mlir-short-SHA>
```

If a `PACKAGE_VENDOR` and `ONNX_MLIR_PRODUCT_VERSION` were specified when the compiler itself was built, use that information. For example:

```
String dump of section '.comment':
  [     0]  GCC: (GNU) 8.5.0 20210514 (Red Hat 8.5.0-15)
  [    2d]  IBM Z Deep Learning Compiler 9.9.9-zdlc
```

### CODE REVIEW

**Comment by hamptonm1** [comment]
So from my testing we get this: `onnx-mlir version 0.3.1 (93579f5c)`

File: src/Version/Version.cpp [hunk 1]
```diff
@@ -67,6 +75,33 @@ std::string getOnnxMlirFullRepositoryVersion(bool toIncludeLLVM) {
...
+  os << PRODUCT_VERSION_PATCH << '-' << PRODUCT_ID;
+#endif
+  return buf;
+}
+
+std::string getOnnxMlirCommitVersion() {
+  std::string buf;
+  llvm::raw_string_ostream os(buf);
+  std::string OnnxMlirPath = getOnnxMlirRepositoryPath();
+  std::string OnnxMlirCommit = getOnnxMlirCommit();
```
Comment (cjvolzka) [inline src/Version/Version.cpp:93]: In the `#if defined(ONNX_MLIR_PRODUCT_VERSION)` path, `OnnxMlirCommit` is never used. Any reason not to move it to be inside of the `#else` and save a call?

Comment (cjvolzka) [inline src/Version/Version.cpp:93]: These aren't classes so shouldn't the names be `onnxMlirPath` and `onnxMlirCommit` (ie lowercase o at the start)?

File: src/Version/Version.cpp [hunk 2]
```diff
@@ -67,6 +75,33 @@ std::string getOnnxMlirFullRepositoryVersion(bool toIncludeLLVM) {
...
+#if defined(ONNX_MLIR_PRODUCT_VERSION)
+  os << getProductFullVersion();
+#else
+  if (!OnnxMlirCommit.empty()) {
+    os << "onnx-mlir version " ONNX_MLIR_VERSION << ' ' << '('
+       << getOnnxMlirCommit() << ')';
```
[the remark this answers is not in the document]
Comment (hamptonm1) [reply src/Version/Version.cpp:99]: The `getOnnxMlirCommit()` is to get the actual commit short hash. We do not call it twice. There is a `getOnnxMlirCommit()` and `getOnnxMlirCommitVersion`. Maybe I can be more descriptive with the names if that helps.

[the remark this answers is not in the document]
Comment (hamptonm1) [reply src/Version/Version.cpp]: Fixed!

File: src/Version/Version.cpp [hunk 3]
```diff
@@ -67,6 +75,33 @@ std::string getOnnxMlirFullRepositoryVersion(bool toIncludeLLVM) {
...
+  os << PRODUCT_VERSION_MINOR << '.';
+  os << PRODUCT_VERSION_PATCH << '-' << PRODUCT_ID;
+#endif
+  return buf;
+}
+
+std::string getOnnxMlirCommitVersion() {
+  std::string buf;
+  llvm::raw_string_ostream os(buf);
+  std::string OnnxMlirPath = getOnnxMlirRepositoryPath();
```
[the remark this answers is not in the document]
Comment (hamptonm1) [reply src/Version/Version.cpp]: Fixed!

File: src/Version/Version.cpp (hunk 1 above)
Comment (hamptonm1) [reply src/Version/Version.cpp]: Updated!

**Comment by hamptonm1** [comment]
Output from the command: 

[meganh@linux1 models]$ readelf -p .comment mnist-12.so
```

String dump of section '.comment':
  [     0]  GCC: (GNU) 8.5.0 20210514 (Red Hat 8.5.0-15)
  [    2d]  IBM Z Deep Learning Compiler 9.9.9-b5a9bc7
```

File: src/Version/Version.cpp (hunk 1 above)
Comment (hamptonm1) [reply src/Version/Version.cpp]: Updated!

File: src/Version/Version.cpp [hunk 4]
```diff
@@ -115,10 +150,9 @@ std::string getOnnxMlirFullVersion(bool toIncludeLLVM) {
 #endif
 
 void getVersionPrinter(llvm::raw_ostream &os) {
-#if defined(ONNX_MLIR_VENDOR) && defined(ONNX_MLIR_PRODUCT_VERSION)
-  os << ONNX_MLIR_VENDOR << " " << PRODUCT_VERSION_MAJOR << '.';
-  os << PRODUCT_VERSION_MINOR << '.';
-  os << PRODUCT_VERSION_PATCH << '-' << PRODUCT_ID << '\n';
+  os << getOnnxMlirCommitVersion() << "\n";
+#if defined(ONNX_MLIR_PRODUCT_VERSION)
+  os << getProductFullVersion() << "\n";
```
[the remark this answers is not in the document]
Comment (hamptonm1) [reply src/Version/Version.cpp]: We discussed :)

**Review by cjvolzka — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**